### PR TITLE
MULE-12068: improvements to use cache in TransformerFactory

### PR DESCRIPTION
**Pull request by alepulver**

### CODE REVIEW

File: modules/xml/src/test/java/org/mule/util/XmlLibrariesVerificationTestCase.java [hunk 1]
```diff
@@ -58,4 +59,11 @@ public void transformer()
         TransformerFactory factory = TransformerFactory.newInstance();
         assertThat(factory, instanceOf(org.apache.xalan.processor.TransformerFactoryImpl.class));
     }
+
+    @Test
+    public void schema()
```
Comment (elrodro83) [inline modules/xml/src/test/java/org/mule/util/XmlLibrariesVerificationTestCase.java]: can you confirm the Validator implementation won't change based on how the classpath is fromed?

File: modules/json/src/main/java/org/mule/module/json/transformers/TransformerInputs.java [hunk 2]
```diff
@@ -96,15 +96,14 @@ public static TransformerFactory createTransformerFactory()
         {
             // Create a factory we know to be STAX-compliant
             transformerFactory = (TransformerFactory) Class.forName(PREFERRED_TRANSFORMATION_FACTORY_CLASS_NAME).newInstance();
+            XMLSecureFactories.createDefault().configureTransformerFactory(transformerFactory);
         }
         catch (Exception ex)
         {
             // Fall back to default factory
-            transformerFactory = TransformerFactory.newInstance();
+            transformerFactory = XMLSecureFactories.createDefault().getTransformerFactory();
```
Comment (elrodro83) [inline modules/json/src/main/java/org/mule/module/json/transformers/TransformerInputs.java]: Have you checked the other modules for occurrences of this code to migrate?

File: modules/xml/src/test/java/org/mule/util/XmlLibrariesVerificationTestCase.java (hunk 1 above)
Comment (alepulver) [inline modules/xml/src/test/java/org/mule/util/XmlLibrariesVerificationTestCase.java]: It is generated from the SchemaLoader instance, so there is a direct dependency.

File: modules/json/src/main/java/org/mule/module/json/transformers/TransformerInputs.java (hunk 2 above)
Comment (alepulver) [inline modules/json/src/main/java/org/mule/module/json/transformers/TransformerInputs.java]: For now, AES has uses of TransformerFactory.

**Pull request merged**